src/main/java/io/strimzi/kafka/bridge/mqtt/core/MqttServerHandler.java

```diff
@@ -73,6 +90,8 @@ protected void channelRead0(ChannelHandlerContext ctx, MqttMessage msg) {
     @Override
     public void exceptionCaught(ChannelHandlerContext ctx, Throwable cause) {
         cause.printStackTrace();
+        this.bridgeKafkaProducerOne.close();
```

Comment:
I mistakenly closed the producer in a wrong callback. 
exceptionCaught(ChannelHandlerContext, Throwable) is invoked when an exception occurs inside a channel (client connection), we can not close the producers if a client caught an exception. 
so I have moved this to the appropriate callback, the one that is invoked when the handler is removed from the pipeline, meaning that it will no longer handler channels (because the server has topped for example)